addons/netfox/rollback/rollback-synchronizer.gd

```diff
@@ -178,36 +181,69 @@ func _before_tick(_delta, tick):
 	var state = _get_history(_states, tick)
 	PropertySnapshot.apply(state, _property_cache)
 
-func _after_tick(_delta, _tick):
+func _after_tick(_delta: float, _tick: int):
 	if not _auth_input_props.is_empty():
-		var input = PropertySnapshot.extract(_auth_input_props)
-		_inputs[NetworkTime.tick] = input
-
-		#Send the last n inputs for each property 
-		var inputs = {}
-		for i in range(0, NetworkRollback.input_redundancy):
-			var tick_input = _inputs.get(NetworkTime.tick - i, {})
-			for property in tick_input:
-				if not inputs.has(property):
-					inputs[property] = []
-				inputs[property].push_back(tick_input[property])
-
-		_attempt_submit_input(inputs)
-	
+		var local_input: Dictionary = PropertySnapshot.extract(_auth_input_props)
+		_inputs[_tick] = local_input
+			
+		if (NetworkRollback.enable_input_serialization):
+			var serialized_current_input: PackedByteArray = PropertiesSerializer.serialize_multiple_properties(_auth_input_props, _tick)
+			_serialized_inputs[_tick] = serialized_current_input
+			
+			if (serialized_inputs_to_send.size() == NetworkRollback.input_redundancy):
+				serialized_inputs_to_send.remove_at(0)
+			serialized_inputs_to_send.append(serialized_current_input)
+			
+			if (serialized_inputs_to_send.is_empty() == false):
+				var merged_serialized_inputs: PackedByteArray
+				merged_serialized_inputs.resize(0)
+				for picked_serialized_input in serialized_inputs_to_send:
+					merged_serialized_inputs.append_array(picked_serialized_input)
+
+				_attempt_submit_serialized_inputs(merged_serialized_inputs)
+		else:
+			#Send the last n inputs for each property
+			var inputs = {}
+			for i in range(0, NetworkRollback.input_redundancy):
+				var tick_input: Dictionary = _inputs.get(_tick - i, {})
+				for property in tick_input:
+					if not inputs.has(property):
+						inputs[property] = []
+					inputs[property].push_back(tick_input[property])
+
+			_attempt_submit_raw_input(inputs)
+		
+	history_cleanup()
+func history_cleanup() -> void:
 	while _states.size() > NetworkRollback.history_limit:
 		_states.erase(_states.keys().min())
 	
 	while _inputs.size() > NetworkRollback.history_limit:
 		_inputs.erase(_inputs.keys().min())
 		
+	if (NetworkRollback.enable_input_serialization):
+		while _serialized_inputs.size() > NetworkRollback.history_limit:
+			_serialized_inputs.erase(_serialized_inputs.keys().min())
+		
 	_freshness_store.trim()
 
-func _attempt_submit_input(input: Dictionary):
+## Sends batched inputs to all other players (not local!)
+func _attempt_submit_raw_input(batched_inputs: Dictionary):
+	# TODO: Default to input broadcast in mesh network setups
+	if enable_input_broadcast:
+		for picked_peer_id in multiplayer.get_peers():
+			_submit_raw_input.rpc_id(picked_peer_id, batched_inputs, NetworkTime.tick)
```

Comment:
This is probably futile (`rpc_id` to all other peers, instead of `rpc`) because @rpc(`call_remote`) does exactly that - calls an RPC to everyone except the invoker. I can confirm doing this at the state is futile since the host is always authority (so #254 ), haven't tested for input but should be the same for input as well.